tests/integration/security/sds_ingress/util/util.go

```diff
@@ -644,3 +644,34 @@ func SetInstances(apps echo.Services) error {
 	}
 	return nil
 }
+
+// Some cloud platform may throw the following error during creation of the service with mixed TCP/UDP protocols:
+// "Error syncing load balancer: failed to ensure load balancer: mixed protocol is not supported for LoadBalancer".
+// Make sure the service is up and running before proceeding with the test.
+func WaitForIngressQUICService(t framework.TestContext, ns string) error {
+	return retry.UntilSuccess(func() error {
+		services, err := t.Clusters().Default().Kube().CoreV1().Services(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(services.Items) == 0 {
+			return fmt.Errorf("still waiting for the service in namespace %s to be created", ns)
+		}
+
+		// Fetch events to check for the service status
+		events, err := t.Clusters().Default().Kube().CoreV1().Events(ns).List(context.TODO(), metav1.ListOptions{})
```

Comment:
Could you filter on `involvedObject.kind`? That way you can iterate over events once and not have to perform a name check

Reply:
Hi @jaellio 

I've applied the `FieldSelector` to the events query with the type - `service` and service `name` field.
It indeed eliminated the need in the services loop.